lib/rspec/matchers/built_in/include.rb

```diff
@@ -51,27 +67,50 @@ def diffable?
 
       private
 
+        def readable_list_of(items)
+          described_items = surface_descriptions_in(items)
+          if described_items.all? { |item| item.is_a?(Hash) }
+            " " + described_items.inject(:merge).to_s
+          else
+            EnglishPhrasing.list(described_items)
+          end
+        end
+
         def invalid_object_message
-          return '' if actual.respond_to?(:include?)
           ", but it does not respond to `include?`"
         end
 
-        def perform_match(predicate, hash_subset_predicate)
-          return false unless actual.respond_to?(:include?)
-
-          expected.__send__(predicate) do |expected_item|
+        def excluded_from_actual(options={ :invert => false })
+          expected.inject([]) do |memo, expected_item|
             if comparing_hash_to_a_subset?(expected_item)
-              expected_item.__send__(hash_subset_predicate) do |(key, value)|
-                actual_hash_includes?(key, value)
+              expected_item.each do |(key, value)|
+                if options[:invert] == true
+                  memo << { key => value } if actual_hash_includes?(key, value)
+                else
+                  memo << { key => value } unless actual_hash_includes?(key, value)
+                end
               end
             elsif comparing_hash_keys?(expected_item)
-              actual_hash_has_key?(expected_item)
+              if options[:invert] == true
+                memo << expected_item if actual_hash_has_key?(expected_item)
+              else
+                memo << expected_item unless actual_hash_has_key?(expected_item)
+              end
             else
-              actual_collection_includes?(expected_item)
+              if options[:invert] == true
+                memo << expected_item if actual_collection_includes?(expected_item)
+              else
+                memo << expected_item unless actual_collection_includes?(expected_item)
+              end
             end
+            memo
```

Comment:
Much better; I knew I could do something with yielding a block but couldn't put it together :/ I'll give that a go.